src/Test/modellike.jl

```diff
@@ -386,6 +386,87 @@ function failcopytestca(dest::MOI.ModelLike)
+    MOI.empty!(model)
+    x, y, z = MOI.add_variables(model, 3)
+    fz = MOI.SingleVariable(z)
+    a = MOI.add_constraint(model, x, MOI.EqualTo(1.0))
+    b = MOI.add_constraint(model, y, MOI.EqualTo(2.0))
+    F1 = MOI.SingleVariable
+    S1 = MOI.EqualTo{Float64}
+    c = MOI.add_constraint(model, MOIU.operate(vcat, Float64, 2.0fz + 1.0), MOI.Nonnegatives(1))
+    F2 = MOI.VectorAffineFunction{Float64}
+    S2 = MOI.Nonnegatives
+
+    vpattr = MOI.VariablePrimalStart()
+    cpattr = MOI.ConstraintPrimalStart()
+    cdattr = MOI.ConstraintDualStart()
+
+    @testset "Newly created variables have start values set to nothing" begin
+        @test !(vpattr in MOI.get(model, MOI.ListOfVariableAttributesSet()))
+        @test MOI.get(model, vpattr, x) === nothing
+        @test MOI.get(model, vpattr, y) === nothing
+        @test MOI.get(model, vpattr, z) === nothing
+
+        @test !(cpattr in MOI.get(model, MOI.ListOfConstraintAttributesSet{F1, S1}()))
+        @test MOI.get(model, cpattr, a) === nothing
+        @test MOI.get(model, cpattr, b) === nothing
+        @test !(cpattr in MOI.get(model, MOI.ListOfConstraintAttributesSet{F2, S2}()))
+        @test MOI.get(model, cpattr, c) === nothing
+
+        @test !(cdattr in MOI.get(model, MOI.ListOfConstraintAttributesSet{F1, S1}()))
+        @test MOI.get(model, cdattr, a) === nothing
+        @test MOI.get(model, cdattr, b) === nothing
+        @test !(cdattr in MOI.get(model, MOI.ListOfConstraintAttributesSet{F2, S2}()))
+        @test MOI.get(model, cdattr, c) === nothing
+    end
+
+    @testset "Allows unsetting by nothing" begin
+        # First set the attributes to some values.
+        MOI.set.((model,), (vpattr,), (x, y, z), (0.0, 1.0, 2.0))
+        MOI.set.((model,), (cpattr,), (a, b), (0.0, 1.0))
+        MOI.set.((model,), (cdattr,), (a, b), (0.0, 1.0))
+        MOI.set(model, cpattr, c, [2.0])
+        MOI.set(model, cdattr, c, [2.0])
+
+        @test vpattr ∈ MOI.get(model, MOI.ListOfVariableAttributesSet())
+        @test cpattr ∈ MOI.get(model, MOI.ListOfConstraintAttributesSet{F1, S1}())
+        @test cpattr ∈ MOI.get(model, MOI.ListOfConstraintAttributesSet{F2, S2}())
+        @test cdattr ∈ MOI.get(model, MOI.ListOfConstraintAttributesSet{F1, S1}())
+        @test cdattr ∈ MOI.get(model, MOI.ListOfConstraintAttributesSet{F2, S2}())
+
+        @test all(MOI.get.((model,), (vpattr,), (x, y, z)) .== (0.0, 1.0, 2.0))
+        @test all(MOI.get.((model,), (cpattr,), (a, b)) .== (0.0, 1.0))
+        @test all(MOI.get.((model,), (cdattr,), (a, b)) .== (0.0, 1.0))
+        @test MOI.get(model, cpattr, c) == [2.0]
+        @test MOI.get(model, cdattr, c) == [2.0]
+
+        MOI.set(model, vpattr, y, nothing)
+        MOI.set(model, cpattr, a, nothing)
+        MOI.set(model, cdattr, a, nothing)
+        MOI.set(model, cpattr, c, [nothing])
+        MOI.set(model, cdattr, c, [nothing])
+
+        #@test cpattr ∉ MOI.get(model, MOI.ListOfConstraintAttributesSet{F2, S2}())
```

Comment:
Why is this commented ?

Reply:
As I said above:

> My new test method has 5 commented @test calls. They fail if uncommented. I am not sure if this is desired behavior or not. I expect a decision from the maintainers about removing or uncommenting them. In the current situation (with these 5 commented) all tests pass.

Reply:
Expanding a little on the above:

The documentation for [`ListOfVariableAttributesSet`](https://jump.dev/MathOptInterface.jl/stable/apireference/#MathOptInterface.ListOfVariableAttributesSet) and for [`ListOfConstraintAttributesSet`](https://jump.dev/MathOptInterface.jl/stable/apireference/#MathOptInterface.ListOfConstraintAttributesSet) are both not very clear (at least to me) on what happens if the attributes are unset.

* If I search for the attributes in these lists from a newly created model, they do not exist in the lists, as it is expected.
* If I search for them in these lists after setting them  in the model, they do exist in the lists, as it is expected.
* The problem is that after *unsetting* the attributes, they keep existing in these lists, I do not know if this is desired behaviour.

More specifically, I do not know if this is a list for every attribute that was ever set (i.e., `MOI.set` was called with it), or just for the attributes with a value different from `nothing` at the moment the list is queried.

Reply:
Yes, this is a bit ambiguous, we don't want the definition to be to constrained here as it would make it difficult to implement and it does not change anything for the main use case which is using it for copying.
You can remove these tests or add a comment explaining that it currently fails but we may want to enable these in the future.

Reply:
I kept them commented and added a comment explaining why they are commented referencing this thread.